plugins/modules/vcenter_ovf_libraryitem.py

```diff
@@ -261,93 +265,85 @@
-    - storage_uri: nfs://datastore.test/srv/share/content-library
-      type: OTHER
-    state: present
-  register: nfs_lib
-
+---
 - name: Export the VM as an OVF on the library
   vmware.vmware_rest.vcenter_ovf_libraryitem:
     session_timeout: 2900
     source:
       type: VirtualMachine
-      id: '{{ my_vm.id }}'
+      id: "{{ lookup('vmware.vmware_rest.vm_moid', 'My VM') }}"
     target:
-      library_id: '{{ nfs_lib.id }}'
+      library_id: 1111111-1111111-1111111-111111
     create_spec:
-      name: golden_image
-      description: an OVF example
+      name: Golden Image
+      description: A golden image for templating
       flags: []
     state: present
-  register: ovf_item
-
-- name: Get the list of items of the NFS library
-  vmware.vmware_rest.content_library_item_info:
-    library_id: '{{ nfs_lib.id }}'
-  register: lib_items
 
-- name: Create a new VM from the OVF
+# Due to an issue with the generated module code and the REST API limitations,
+# updating existing template descriptions must be done by removing the create_spec.name
+# attribute.
+#
+# Note that omitting the create_spec.name attribute will force the OVF template to be replaced.
+# Thus, the second task will always report as changed
+- name: Export the VM as an OVF on the library
   vmware.vmware_rest.vcenter_ovf_libraryitem:
-    ovf_library_item_id: '{{ (lib_items.value|selectattr("name", "equalto", "golden_image")|first).id }}'
-    state: deploy
+    session_timeout: 2900
+    source:
+        type: VirtualMachine
+        id: "{{ lookup('vmware.vmware_rest.vm_moid', 'My VM') }}"
     target:
-      resource_pool_id: "{{ lookup('vmware.vmware_rest.resource_pool_moid', '/my_dc/host/my_cluster/Resources') }}"
-    deployment_spec:
-      name: my_vm_from_ovf
-      accept_all_EULA: true
-      storage_provisioning: thin
+        library_id: 1111111-1111111-1111111-111111
+    create_spec:
+        name: Golden Image
+        description: Some unhelpful description
+        flags: []
+    state: present
+  register: _ovf_item
 
-- name: Create a new VM from the OVF and specify the host and folder
+- name: Update The Template Description
   vmware.vmware_rest.vcenter_ovf_libraryitem:
-    ovf_library_item_id: '{{ (lib_items.value|selectattr("name", "equalto", "golden_image")|first).id }}'
-    state: deploy
+    session_timeout: 2900
+    source:
+      type: VirtualMachine
+      id: "{{ lookup('vmware.vmware_rest.vm_moid', vm_name) }}"
     target:
-      resource_pool_id: "{{ lookup('vmware.vmware_rest.resource_pool_moid', '/my_dc/host/my_cluster/Resources') }}"
-      folder_id: "{{ lookup('vmware.vmware_rest.folder_moid', '/my_dc/vm') }}"
-      host_id: "{{ lookup('vmware.vmware_rest.host_moid', '/my_dc/host/my_cluster/esxi1.test/test_vm1') }}"
-    deployment_spec:
-      name: my_vm_from_ovf_on_a_host
-      accept_all_EULA: true
-      storage_provisioning: thin
+      library_id: "{{ library_id }}"
+      library_item_id: "{{ _ovf_item.id }}"
```

Comment:
according to the real result
```suggestion
      library_item_id: "{{ _ovf_item.ovf_library_item_id }}"
```

Reply:
`ovf_library_item_id` is only returned if the item already exists. If the item does not exist, the `id` attribute is returned. So i think the original is correct in this example